mlflow/transformers/__init__.py

```diff
@@ -2670,10 +2678,55 @@ def is_uri(s):
         if not valid_uri:
             raise MlflowException(
                 "An invalid string input was provided. String inputs to "
-                "audio files must be either a file location or a uri.",
+                "audio or image files must be either a file location or a uri.",
                 error_code=BAD_REQUEST,
             )
 
+    def _convert_image_input(self, data):
+        """
+        Conversion utility for decoding the base64 encoded bytes data of a raw imagefile when
+        parsed through model serving, if applicable. Direct usage of the pyfunc implementation
+        outside of model serving will treat this utility as a noop.
+
+        For reference, the expected encoding for input to Model Serving will be:
+
+        import requests
+        import base64
+
+        response = requests.get("http://images.cocodataset.org/val2017/000000039769.jpg")
+        encoded_image = base64.b64encode(response.content).decode("ascii")
+
+        inference_data = json.dumps({"inputs": [encoded_image]})
+
+        or
+
+        inference_df = pd.DataFrame(
+        pd.Series([encoded_image], name="image_file")
+        )
+        split_dict = {"dataframe_split": inference_df.to_dict(orient="split")}
+        split_json = json.dumps(split_dict)
+
+        or
+
+        records_dict = {"dataframe_records": inference_df.to_dict(orient="records")}
+        records_json = json.dumps(records_dict)
+
+        This utility will convert this JSON encoded, base64 encoded text back into bytes for
+        input into the imageclassificationPipeline for inference.
+        """
+        if isinstance(data, list) and all(isinstance(element, dict) for element in data):
+            images = []
+            for item in data:
+                encoded_image = next(iter(item.values()))
+                if isinstance(encoded_image, str):
+                    self._validate_str_input_uri_or_file(encoded_image)
+                images.append(decode_audio(encoded_image))  # noqa: F821
```

Comment:
This won't work, `decode_audio` function is defined inside `_convert_audio_input`, could you extract that function out and rename it?